crates/router/src/connector/expresscheckout/transformers.rs

```diff
@@ -0,0 +1,363 @@
+    deserializer: D,
+) -> Result<ExpresscheckoutPaymentStatus, D::Error>
+where
+    ExpresscheckoutPaymentStatus: Default + Deserialize<'de>,
+    D: Deserializer<'de>,
+{
+    let opt = ExpresscheckoutPaymentStatus::deserialize(deserializer);
+    match opt {
+        Ok(v) => Ok(v),
+        _ => Ok(ExpresscheckoutPaymentStatus::default()),
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct Authentication {
+    authentication: AuthenticationData,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct AuthenticationData {
+    url: String,
+    method: Method,
+}
+
+impl<F, T>
+    TryFrom<
+        types::ResponseRouterData<
+            F,
+            ExpresscheckoutPaymentsResponse,
+            T,
+            types::PaymentsResponseData,
+        >,
+    > for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ParsingError>;
+    fn try_from(
+        item: types::ResponseRouterData<
+            F,
+            ExpresscheckoutPaymentsResponse,
+            T,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            status: enums::AttemptStatus::from(item.response.status),
+            response: Ok(types::PaymentsResponseData::TransactionResponse {
+                resource_id: item.response.txn_uuid.map_or(
+                    types::ResponseId::NoResponseId,
+                    types::ResponseId::ConnectorTransactionId,
+                ),
+                redirection_data: item.response.payment.map(|r| services::RedirectForm {
+                    url: r.authentication.url,
+                    method: r.authentication.method,
+                    form_fields: HashMap::new(),
+                }),
+                redirect: true,
+                mandate_reference: None,
+                connector_metadata: None,
+            }),
+            ..item.data
+        })
+    }
+}
+
+#[derive(Default, Debug, Serialize)]
+pub struct ExpresscheckoutRefundRequest {
+    unique_request_id: String,
+    amount: String,
+}
+
+impl<F> TryFrom<&types::RefundsRouterData<F>> for ExpresscheckoutRefundRequest {
+    type Error = error_stack::Report<errors::ParsingError>;
+    fn try_from(item: &types::RefundsRouterData<F>) -> Result<Self, Self::Error> {
+        let refund_req = Self {
+            unique_request_id: item.request.refund_id.clone(),
+            amount: item.request.amount.to_string(),
+        };
+        Ok(refund_req)
+    }
+}
+
+// Type definition for Refund Response
+
+#[allow(dead_code)]
+#[derive(Debug, Serialize, Default, Deserialize, Clone)]
+pub enum RefundStatus {
+    #[serde(rename = "ERROR")]
+    #[default]
+    Error,
+    #[serde(rename = "BAD REQUEST")]
+    BadRequest,
+    #[serde(rename = "NOT_FOUND")]
+    NotFound,
+    #[serde(rename = "SUCCESS")]
+    Success,
+    #[serde(rename = "PENDING")]
```

Comment:
Instead of renaming every status into upper case, rename where ever necessary. In our case BadRequest and NotFound. For rest all remove rename and add `#[serde(rename_all = "UPPERCASE")]` to the enum itself